src/lean_spec/subspecs/containers/state/state.py

```diff
@@ -398,11 +415,18 @@ def process_block(self, block: Block) -> "State":
         state = self.process_block_header(block)
 
         # Process justification votes (attestations).
-        return state.process_attestations(block.body.attestations)
+        raw_attestations: List[
+            Union[ValidatorAttestation, ProposerAttestationData]
```

Comment:
we don't need union here, lets convert ProposerAttestationData to the proposer's ProposerAttestationData (take target, source from there, head is the hashtreeroot of the current block and slot) and append 

(something that I missed doing that properly in markdown)